packages/graphql-yoga/__tests__/accept-header.spec.ts

```diff
@@ -0,0 +1,138 @@
+import { ExecutionResult } from 'graphql'
+import { createSchema, createYoga, Repeater, Plugin } from 'graphql-yoga'
+
+describe('accept header', () => {
+  it('instruct server to return an event-stream', async () => {
+    const yoga = createYoga({
+      schema: createSchema({
+        typeDefs: /* GraphQL */ `
+          type Query {
+            ping: String
+          }
+        `,
+      }),
+    })
+
+    // NOTE: this fails with content-type: application/json and body instead of query params
+    // is that intended?
+    const response = await yoga.fetch(`http://yoga/graphql?query=query{ping}`, {
+      headers: {
+        accept: 'text/event-stream',
+      },
+    })
+    expect(response.headers.get('content-type')).toEqual('text/event-stream')
+    const iterator = response.body![Symbol.asyncIterator]()
+    const { value } = await iterator.next()
+    const valueStr = Buffer.from(value).toString('utf-8')
+    expect(valueStr).toContain(
+      `data: ${JSON.stringify({ data: { ping: null } })}`,
+    )
+  })
+
+  it('instruct server to return a multipart result', async () => {
+    const yoga = createYoga({
+      schema: createSchema({
+        typeDefs: /* GraphQL */ `
+          type Query {
+            ping: String
+          }
+        `,
+        resolvers: {
+          Query: { ping: () => 'pong' },
+        },
+      }),
+    })
+
+    // NOTE: this fails with content-type: application/json and body instead of query params
+    // is that intended?
```

Comment:
Should we also add test cases for this scenario, I discovered this randomly (**unrelated to other comment in this test file**)

Reply:
Yes because if there are query params, it should be GET and GET requests cannot have body.